app/models/spree/gateway/amazon.rb

```diff
@@ -68,11 +68,17 @@ def purchase(amount, amazon_checkout, gateway_options={})
       capture(amount, amazon_checkout, gateway_options)
     end
 
-    def credit(amount, _credit_card, gateway_options={})
-      order = Spree::Order.find_by(:number => gateway_options[:order_id].split("-")[0])
-      load_amazon_mws(order.amazon_order_reference_id)
-      capture_id = order.amazon_transaction.capture_id
-      response = @mws.refund(capture_id, gateway_options[:order_id], amount / 100.00, Spree::Config.currency)
+    def credit(amount, _response_code, gateway_options)
```

Comment:
Ah good catch, if I make them optional and they're not passed in it will still fail as `gateway_options[:originator].payment` will fail. I think we should preserve the method signature as spree has it, with optional gateway_options, I'll make the change thanks!